Incorrect color tags when converting a PAL DVD to mp4

### Problem Description

When converting a VIDEO_TS folder of a PAL DVD to an mp4 video file, the resulting video should have the (PAL) color profile: **5-1-6**, but has the color profile: **5-5-5** (at least in my case). As a consequence, the colors of the video in **QuickTime Player** are off.

This can be fixed with the extra video options
`videoformat=pal:colorprim=bt470bg:transfer=bt709:colormatrix=smpte170m`
but of course, this shouldn’t be necessary.


### Activity Log, Crash Log or any other details

```text
HandBrake Activity Log for Session: 2021-10-31T20:30:24Z
Handbrake Version: 1.4.2 (2021100300)
UVWXYZ.mp4
Preset: Fast 1080p30
[21:30:24] macgui: fr.handbrake.HandBrakeXPCService scanning specifically for title: 1
[21:30:24] CPU: Intel(R) Xeon(R) CPU           W3680  @ 3.33GHz
[21:30:24]  - logical processor count: 12
[21:30:24] hb_scan: path=/Volumes/Video/Film/UVWXYZ, title_index=1
disc.c:437: error opening file BDMV/index.bdmv
disc.c:437: error opening file BDMV/BACKUP/index.bdmv
bluray.c:2646: nav_get_title_list(/Volumes/Video/Film/UVWXYZ/) failed
[21:30:24] bd: not a bd - trying as a stream/file instead
libdvdread: Couldn't find device name.
libdvdnav: Can't read name block. Probably not a DVD-ROM device.
libdvdread: Couldn't find device name.
[21:30:24] scan: DVD has 5 title(s)
[21:30:24] scan: scanning title 1
[21:30:24] scan: duration is 01:43:52 (6232880 ms)
[21:30:24] pgc_id: 1, pgn: 1: pgc: 0x7fbd3b8018b0
[21:30:24] scan: checking audio 1
[21:30:24] scan: id=0x80bd, lang=Deutsch (AC3), 3cc=deu ext=0
[21:30:24] scan: title 1 has 13 chapters
[21:30:24] scan: chap 1, 82680 ms
[21:30:24] scan: chap 2, 448240 ms
[21:30:24] scan: chap 3, 568360 ms
[21:30:24] scan: chap 4, 690560 ms
[21:30:24] scan: chap 5, 516360 ms
[21:30:24] scan: chap 6, 676520 ms
[21:30:24] scan: chap 7, 328640 ms
[21:30:24] scan: chap 8, 710840 ms
[21:30:24] scan: chap 9, 499200 ms
[21:30:24] scan: chap 10, 433160 ms
[21:30:24] scan: chap 11, 592280 ms
[21:30:24] scan: chap 12, 544440 ms
[21:30:24] scan: chap 13, 141600 ms
[21:30:24] scan: aspect = 4:3
[21:30:24] scan: decoding previews for title 1
[21:30:24] scan: title angle(s) 1
[21:30:24] scan: audio 0x80bd: ac3, rate=48000Hz, bitrate=384000 Deutsch (AC3) (5.1 ch) (384 kbps)
[21:30:24] scan: 10 previews, 720x576, 25.000 fps, autocrop = 2/2/8/12, aspect 4:3, PAR 16:15, color profile: 5-5-5
[21:30:24] libhb: scan thread found 1 valid title(s)
[21:30:24] macgui: fr.handbrake.HandBrakeXPCService scan done
[21:30:24] macgui: fr.handbrake.HandBrakeXPCService started encoding UVWXYZ.mp4
[21:30:24] macgui: fr.handbrake.HandBrakeXPCService with preset Fast 1080p30
[21:30:24] Starting work at: Sun Oct 31 21:30:24 2021

[21:30:24] 1 job(s) to process
[21:30:24] Starting Task: Encoding Pass
[21:30:24] job configuration:
[21:30:24]  * source
[21:30:24]    + /Volumes/Video/Film/UVWXYZ
[21:30:24]    + title 1, chapter(s) 1 to 13
[21:30:24]  * destination
[21:30:24]    + /Volumes/Media/UVWXYZ.mp4
[21:30:24]    + container: MPEG-4 (libavformat)
[21:30:24]      + align initial A/V stream timestamps
[21:30:24]      + chapter markers
[21:30:24]  * video track
[21:30:24]    + decoder: mpeg2video 8-bit (yuv420p)
[21:30:24]      + bitrate 200 kbps
[21:30:24]    + filters
[21:30:24]      + Comb Detect (mode=3:spatial-metric=2:motion-thresh=1:spatial-thresh=1:filter-mode=2:block-thresh=40:block-width=16:block-height=16)
[21:30:24]      + Decomb (mode=39)
[21:30:24]      + Framerate Shaper (mode=2:rate=27000000/900000)
[21:30:24]        + frame rate: 25.000 fps -> peak rate limited to 30.000 fps
[21:30:24]      + Crop and Scale (width=700:height=572:crop-top=2:crop-bottom=2:crop-left=8:crop-right=12)
[21:30:24]        + source: 720 * 576, crop (2/2/8/12): 700 * 572, scale: 700 * 572
[21:30:24]    + Output geometry
[21:30:24]      + storage dimensions: 700 x 572
[21:30:24]      + pixel aspect ratio: 16 : 15
[21:30:24]      + display dimensions: 746 x 572
[21:30:24]    + encoder: H.264 (libx264)
[21:30:24]      + preset:  fast
[21:30:24]      + profile: main
[21:30:24]      + level:   4.0
[21:30:24]      + quality: 22.00 (RF)
[21:30:24]      + color profile: 5-5-5
[21:30:24]  * audio track 1
[21:30:24]    + decoder: Deutsch (AC3) (5.1 ch) (384 kbps) (track 1, id 0x80bd)
[21:30:24]      + bitrate: 384 kbps, samplerate: 48000 Hz
[21:30:24]    + mixdown: Stereo
[21:30:24]    + encoder: AAC (Apple AudioToolbox)
[21:30:24]      + bitrate: 160 kbps, samplerate: 48000 Hz
libdvdread: Couldn't find device name.
libdvdnav: Can't read name block. Probably not a DVD-ROM device.
libdvdread: Couldn't find device name.
[21:30:24] sync: expecting 155822 video frames
[21:30:24] encx264: encoding at constant RF 22.000000
[21:30:24] encx264: unparsed options: level=4.0:ref=2:8x8dct=0:weightp=1:subme=6:vbv-bufsize=25000:vbv-maxrate=20000:rc-lookahead=30
x264 [info]: using SAR=16/15
x264 [info]: using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2
x264 [info]: profile Main, level 4.0, 4:2:0, 8-bit
[21:30:25] sync: first pts video is 0
[21:30:25] sync: "Chapter 1" (1) at frame 1 time 0
[21:30:25] sync: first pts audio 0x80bd is 0
[21:30:38] sync: "Chapter 2" (2) at frame 2068 time 7441200
[21:32:07] sync: "Chapter 3" (3) at frame 13274 time 47782800
[21:33:50] sync: "Chapter 4" (4) at frame 27483 time 98935200
[21:36:02] sync: "Chapter 5" (5) at frame 44747 time 161085600
[21:37:38] sync: "Chapter 6" (6) at frame 57656 time 207558000
[21:39:47] sync: "Chapter 7" (7) at frame 74569 time 268444800
[21:40:50] sync: "Chapter 8" (8) at frame 82785 time 298022400
[21:42:55] sync: "Chapter 9" (9) at frame 100556 time 361998000
[21:44:24] sync: "Chapter 10" (10) at frame 113036 time 406926000
[21:45:46] sync: "Chapter 11" (11) at frame 123865 time 445910400
[21:47:37] sync: "Chapter 12" (12) at frame 138672 time 499215600
[21:49:27] sync: "Chapter 13" (13) at frame 152283 time 548215200
[21:49:46] reader: done. 1 scr changes
[21:49:46] work: average encoding speed for job is 134.117996 fps
[21:49:46] comb detect: heavy 2415 | light 6774 | uncombed 146633 | total 155822
[21:49:46] decomb: deinterlaced 2415 | blended 6774 | unfiltered 146633 | total 155822
[21:49:46] vfr: 155822 frames output, 0 dropped and 0 duped for CFR/PFR
[21:49:46] vfr: lost time: 0 (0 frames)
[21:49:46] vfr: gained time: 0 (0 frames) (0 not accounted for)
[21:49:46] ac3-decoder done: 194775 frames, 0 decoder errors
[21:49:46] mpeg2video-decoder done: 155822 frames, 0 decoder errors
[21:49:46] sync: got 155822 frames, 155822 expected
[21:49:46] sync: framerate min 25.000 fps, max 25.000 fps, avg 25.000 fps
x264 [info]: frame I:1566  Avg QP:19.87  size: 31712
x264 [info]: frame P:43623 Avg QP:22.39  size:  7956
x264 [info]: frame B:110633 Avg QP:23.82  size:  2423
x264 [info]: consecutive B-frames:  4.8%  0.9%  1.8% 92.5%
x264 [info]: mb I  I16..4: 26.4%  0.0% 73.6%
x264 [info]: mb P  I16..4:  4.4%  0.0%  6.3%  P16..4: 45.0% 17.2%  9.6%  0.0%  0.0%    skip:17.5%
x264 [info]: mb B  I16..4:  4.0%  0.0%  0.6%  B16..8: 25.7%  6.7%  0.3%  direct:12.9%  skip:49.8%  L0:41.0% L1:50.2% BI: 8.9%
x264 [info]: coded y,uvDC,uvAC intra: 46.3% 62.5% 16.2% inter: 10.7% 26.1% 0.3%
x264 [info]: i16 v,h,dc,p: 37% 19% 19% 24%
x264 [info]: i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 26% 15% 21%  7%  7%  8%  6%  6%  4%
x264 [info]: i8c dc,h,v,p: 59% 16% 21%  4%
x264 [info]: Weighted P-Frames: Y:8.3% UV:2.3%
x264 [info]: ref P L0: 64.6% 35.4%
x264 [info]: ref B L0: 82.9% 17.1%
x264 [info]: ref B L1: 95.3%  4.7%
x264 [info]: kb/s:853.28
[21:49:46] mux: track 0, 155822 frames, 664737784 bytes, 853.20 kbps, fifo 4096
[21:49:46] mux: track 1, 292165 frames, 128092743 bytes, 164.41 kbps, fifo 8192
[21:49:46] Finished work at: Sun Oct 31 21:49:46 2021

[21:49:46] libhb: work result = 0
```


### What Operating System are you running?

macOS 10.14.6

### What version of HandBrake are you running?

Version 1.4.2 (2021100300)

### Where did you download HandBrake from?

handbrake.fr

--- a/libhb/scan.c
+++ b/libhb/scan.c
@@ -1320,12 +1320,20 @@ skip_preview:
         }
         title->pix_fmt = vid_info.pix_fmt;
 
-        if ((title->color_prim     != HB_COLR_PRI_UNDEF &&
-             title->color_prim     != HB_COLR_PRI_UNSET) ||
-            (title->color_transfer != HB_COLR_TRA_UNDEF &&
-             title->color_transfer != HB_COLR_TRA_UNSET) ||
-            (title->color_matrix   != HB_COLR_MAT_UNDEF &&
-             title->color_matrix != HB_COLR_MAT_UNSET))
+        // DVD-Video have no defined color info, but some mostly wrong
+        // values could be read from the mpeg-2 bitstream. Override those here.
+        if (data->dvd)
+        {
+            title->color_prim     = get_color_prim(HB_COLR_PRI_UNSET, vid_info.geometry, vid_info.rate);
+            title->color_transfer = HB_COLR_TRA_BT709;
+            title->color_matrix   = HB_COLR_MAT_SMPTE170M;
+        }
+        else if ((title->color_prim     != HB_COLR_PRI_UNDEF &&
+                  title->color_prim     != HB_COLR_PRI_UNSET) ||
+                 (title->color_transfer != HB_COLR_TRA_UNDEF &&
+                  title->color_transfer != HB_COLR_TRA_UNSET) ||
+                 (title->color_matrix   != HB_COLR_MAT_UNDEF &&
+                  title->color_matrix != HB_COLR_MAT_UNSET))
         {
             title->color_prim     = get_color_prim(title->color_prim, vid_info.geometry, vid_info.rate);
             title->color_transfer = get_color_transfer(title->color_transfer);

